core/core-impl/src/main/java/ai/timefold/solver/core/impl/localsearch/DefaultLocalSearchPhaseFactory.java

```diff
@@ -191,18 +198,112 @@ private UnionMoveSelectorConfig determineDefaultMoveSelectorConfig(HeuristicConf
                 .filter(variableDescriptor -> !variableDescriptor.isListVariable())
                 .distinct()
                 .toList();
+        var hasChainedVariable = basicVariableDescriptorList.stream()
+                .filter(v -> v instanceof BasicVariableDescriptor<Solution_>)
+                .anyMatch(v -> ((BasicVariableDescriptor<?>) v).isChained());
         var listVariableDescriptor = solutionDescriptor.getListVariableDescriptor();
         if (basicVariableDescriptorList.isEmpty()) { // We only have the one list variable.
-            return new UnionMoveSelectorConfig()
-                    .withMoveSelectors(new ListChangeMoveSelectorConfig(), new ListSwapMoveSelectorConfig());
+            if (configPolicy.getNearbyDistanceMeterClass() == null) {
+                return new UnionMoveSelectorConfig()
+                        .withMoveSelectors(new ListChangeMoveSelectorConfig(), new ListSwapMoveSelectorConfig());
+            } else {
+                return new UnionMoveSelectorConfig()
+                        .withMoveSelectors(new ListChangeMoveSelectorConfig(),
+                                new ListSwapMoveSelectorConfig(),
+                                new ListChangeMoveSelectorConfig()
+                                        .withValueSelectorConfig(new ValueSelectorConfig()
+                                                .withId("changeMoveSelector"))
+                                        .withDestinationSelectorConfig(new DestinationSelectorConfig()
+                                                .withNearbySelectionConfig(new NearbySelectionConfig()
+                                                        .withOriginValueSelectorConfig(new ValueSelectorConfig()
+                                                                .withMimicSelectorRef("changeMoveSelector"))
+                                                        .withNearbyDistanceMeterClass(
+                                                                configPolicy.getNearbyDistanceMeterClass()))),
+                                new ListSwapMoveSelectorConfig()
+                                        .withValueSelectorConfig(new ValueSelectorConfig()
+                                                .withId("swapMoveSelector"))
+                                        .withSecondaryValueSelectorConfig(new ValueSelectorConfig()
+                                                .withNearbySelectionConfig(new NearbySelectionConfig()
+                                                        .withOriginValueSelectorConfig(new ValueSelectorConfig()
+                                                                .withMimicSelectorRef("swapMoveSelector"))
+                                                        .withNearbyDistanceMeterClass(
+                                                                configPolicy.getNearbyDistanceMeterClass()))),
+                                new KOptListMoveSelectorConfig()
```

Comment:
I'm asking myself... if we're adding this move, why only for nearby? Does it make sense to add it for both?